File: Programmers/Lv1/최소직사각형.py
def solution(sizes):
    answer = 0

    w = []
    h = []

    n = len(sizes)
    for i in range(n):
        if sizes[i][0] > sizes[i][1]:
            w.append(sizes[i][0])
            h.append(sizes[i][1])
        else:
            h.append(sizes[i][0])
            w.append(sizes[i][1])

    answer = max(w) * max(h)
    # 최대한 작게


    return answer

print('TC1')
print(solution([[60, 50], [30, 70], [60, 30], [80, 40]]))
print('TC2')
print(solution([[10, 7], [12, 3], [8, 15], [14, 7], [5, 15]]))

File: Programmers/Lv1/최소직사각형.js
function solution(sizes) {
  for (size of sizes) {
    size.sort((a, b) => b - a);
  }
  var maxW = sizes.reduce((max, item) => (item[0] > max ? item[0] : max), 0);
  var maxH = sizes.reduce((max, item) => (item[1] > max ? item[1] : max), 0);
  return maxW * maxH;
}
